rust/src/state_machine/phases/mod.rs

```diff
@@ -96,6 +159,43 @@ where
     }
 }
 
+impl<R, S> PhaseState<R, S>
+where
+    Self: Phase<R> + Purge<R>,
+{
+    /// Run the current phase to completion, then transition to the
+    /// next phase and return it.
+    pub async fn run_phase(mut self) -> Option<StateMachine<R>> {
+        if let Err(err) = self.run().await {
+            return Some(self.into_error_state(err));
+        } else {
```

Comment:
Was it intended to leave the "else" here?

Reply:
Somehow I thought rustc would reject the code without the `else` but it seems accepted. I'll remove the `else`.